### PR TITLE
Rahav/cert 1850 external documentation

### PR DESCRIPTION
**Pull request by rahav-certora**

### CODE REVIEW

File: docs/prover/cli/options.md [hunk 1]
```diff
@@ -258,12 +255,15 @@ Whenever you want to use a Solidity compiler executable with a non-default name.
 
 **What does it do?**  
 Gets a list of arguments to pass to the Solidity compiler. The arguments will be passed as is, without any formatting, in the same order.  
+arguments that are defined as standalone flags cannot be defined in `--solc_args`, including the solc flags  `--optimize`,
```
Comment (mdgeorge4153) [inline docs/prover/cli/options.md:258]: `solc` fails spell checking; you should either use "the Solidity compiler" or
```
`solc`
```

**Review by mdgeorge4153 — DISMISSED** [review]
This currently doesn't compile; it looks like there are missing labels for `--staging` and `--cloud` and missing code annotations in a few places.  See the "checks have failed" section below for more details

File: docs/prover/cli/options.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ When you wish to prove properties on the source code. This is by far the most co
 If we have a Solidity file `Bank.sol`, with a contract named `Bank` inside it, and a specification file called `Bank.spec`, the run command would be:  
 `certoraRun Bank.sol --verify Bank:Bank.spec`
 
-### `--assert`
+### `--assert_contract`
```
Comment (shellygr) [inline docs/prover/cli/options.md:40]: isn't that assert_contracts? I'm always confusing this one

Comment (rahav-certora) [reply docs/prover/cli/options.md:40]: There is assert_contracts but we marked is as "internal" so it should not be in this doc

Comment (rahav-certora) [reply docs/prover/cli/options.md:40]: fixed (apparently  you are not the only one that get confused here)

File: docs/prover/cli/options.md (hunk 1 above)
Comment (rahav-certora) [reply docs/prover/cli/options.md:258]: done

**Comment by rahav-certora** [comment]
regarding --cloud and --staging:
I thought we only document flags that are external, --cloud and --staging are not. This document from the start was not complete. This PR was only to remove references to deprecated flags

File: docs/prover/cli/options.md [hunk 3]
```diff
@@ -276,17 +276,63 @@ When different contracts have to be compiled for different Solidity versions.
...
+
+### `--solc_allow_path`
 
 **What does it do?**  
 Passes the value of this option as is to the solidity compiler's option `--allow-paths`.
 See [--allow-path specification](https://docs.soliditylang.org/en/v0.8.16/path-resolution.html#allowed-paths)
 
 **When to use it?**  
-When we want for security reasons to limit the locations for loaded sources to specific directories
+When we want all an additional location for loading sources
```
Comment (shellygr) [inline docs/prover/cli/options.md:331]:
```suggestion
When we want to add an additional location the Solidity compiler to load sources from
```

File: docs/prover/cli/options.md [hunk 4]
```diff
@@ -646,39 +650,28 @@ Upon instruction from the Certora team.
...
-(--settings)=
-### `--settings`
+(--prover_args)=
+### `--prover_args`
 
-The `--settings` option allows you to provide fine-grained tuning options to the
-Prover.  `--settings` should be followed by a comma-separated list of options.
+The `--prover_args` option allows you to provide fine-grained tuning options to the
+Prover.  `--prover_args` receives a string containing prover flags, the prover flags will be sent as it to the prover.
+`--prover_args` cannot set prover options that are set by standalone certoraRun options (e.g. the prover option`--t` is
```
Comment (shellygr) [inline docs/prover/cli/options.md:660]:
```suggestion
`--prover_args` cannot set prover options that are set by standalone certoraRun options (e.g. the prover option `--t` is
```

File: docs/prover/cli/options.md [hunk 5]
```diff
@@ -646,39 +650,28 @@ Upon instruction from the Certora team.
...
+set by `--smt_timeout` therefore cannot appear in `--prover_args`). `--prover_args` value must be quoted
```
Comment (shellygr) [inline docs/prover/cli/options.md:661]:
```suggestion
set by `--smt_timeout` therefore cannot appear in `--prover_args`). `--prover_args` value must be quoted.
```

**Review by shellygr — APPROVED** [review]
looks better to me - just a few suggestions

**Comment by mdgeorge4153** [comment]
> regarding --cloud and --staging: I thought we only document flags that are external, --cloud and --staging are not. This document from the start was not complete. This PR was only to remove references to deprecated flags

You can see the error log by clicking on the details; if you get a 404 you need to log into readthedocs.com first.  Here are the errors:
```
.../docs/cvl/cvl2/migration.md:88: WARNING: undefined label: --settings
.../docs/gambit/mutation-verifier.md:96: WARNING: undefined label: --staging
.../docs/gambit/mutation-verifier.md:96: WARNING: undefined label: --cloud
```

I suspect the migration guide says something like "the --settings flags have changed" in which case the ref can be removed (or maybe updated to explain what it's replaced with); I'm not sure why the gambit docs are referring to these settings; if it's not clear from context, ping Chandra

**Review by mdgeorge4153 — APPROVED** [review]

File: docs/prover/cli/options.md [hunk 6]
```diff
@@ -646,39 +650,28 @@ Upon instruction from the Certora team.
...
 
 This option determines whether {ref}`havoc summaries <havoc-summary>` assume
 that the called method returns the correct number of return values.
 
 (-showInternalFunctions)=
-#### `--settings -showInternalFunctions`
+#### `--prover_args '-showInternalFunctions'`
+
+A single occurrence of `--prover_args` can set multiple values
+#### `--prover_args '-showInternalFunctions -optimisticReturnsize'`
```
[the remark this answers is not in the document]
Comment (rahav-certora) [reply docs/prover/cli/options.md:674]: I do not understand what header, I just added additional example

File: docs/prover/cli/options.md [hunk 7]
```diff
@@ -646,39 +650,28 @@ Upon instruction from the Certora team.
...
-
-`certoraRun Bank.sol --verify Bank:bank.spec --settings -saveRerunData`
-
-`certoraRun Bank.sol --verify Bank:bank.spec --rerun_verification rerun_checkBank.rerunbin`
+`--java_args '"-Dcvt.default.parallelism=2"'` - will set the number of “tasks” that can run in parallel to 2.
 
-(--settings)=
-### `--settings`
+(--prover_args)=
+### `--prover_args`
```
Comment (naftali-g) [inline docs/prover/cli/options.md:656]: this is the header you changed, and more importantly notice that in line 655 you changed the string used to reference this header from other files.

File: docs/prover/cli/options.md [hunk 8]
```diff
@@ -596,45 +632,13 @@ When you suspect you have an old installation. To install the newest version, us
...
-If you want only to check your spec, or include it in an automated task (e.g., a git `pre-commit` hook).  
-**Example**
 
-`certoraRun Bank.sol --verify Bank:bank.spec --typecheck_only`
 
 Advanced options
 ----------------
 
-(--cloud)=
-### `--cloud`
```
Comment (naftali-g) [inline docs/prover/cli/options.md:614]: You removed this header and its reference, so all usages of that reference ned to be updated.

File: docs/prover/cli/options.md [hunk 9]
```diff
@@ -596,45 +632,13 @@ When you suspect you have an old installation. To install the newest version, us
...
-Runs the Prover on the cloud.  Note that for non-Certora users, `--cloud` is
-the default, so this option does nothing.
-
-**When to use it?**
-
-If you are a Certora employee who usually runs the Prover locally, but want to
-run on the cloud instead.
-
-(--staging)=
-### `--staging [branch]`
```
Comment (naftali-g) [inline docs/prover/cli/options.md:627]: ditto

**Pull request merged**